evals/cli/oaieval.py

```diff
@@ -49,12 +47,40 @@ def get_parser() -> argparse.ArgumentParser:
         help="Path to the registry",
     )
     parser.add_argument("--debug", action=argparse.BooleanOptionalAction, default=False)
-    parser.add_argument("--local-run", action=argparse.BooleanOptionalAction, default=True)
+    parser.add_argument(
+        "--local-run",
+        action=argparse.BooleanOptionalAction,
+        default=True,
+        help="Enable local mode for running evaluations. In this mode, the evaluation results are stored locally in a JSON file. This mode is enabled by default."
+    )
+
+    parser.add_argument(
+        "--http-run",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+        help="Enable HTTP mode for running evaluations. In this mode, the evaluation results are sent to a specified URL rather than being stored locally or in Snowflake. This mode should be used in conjunction with the '--http-run-url' and '--http-batch-size' arguments."
+    )
+
+    parser.add_argument(
+        "--http-run-url", 
+        type=str, 
+        default=None,
+        help="URL to send the evaluation results when in HTTP mode. This option should be used in conjunction with the '--http-run' flag."
+    )
+
+    parser.add_argument(
+        "--http-batch-size", 
+        type=int, 
+        default=1,
```

Comment:
We should make `default=None` and when http_batch_size is `None`, don't limit the size of the batches. I think most users will generally prefer lower overhead / a bigger batch size.

Reply:
I appreciate your viewpoint on making `default=None` and not limiting the batch size when `http_batch_size` is None. The idea of having lower overhead and a larger batch size for improved efficiency is a valid one.

However, let's consider a situation where someone wants to export 10,000 evaluations(which I regularly do). Given that a single `abstract_reasoning` evaluation is approximately 1KB, the total size of the request would be around 10MB.

<img width="1314" alt="Screenshot 2023-07-25 at 7 41 58 PM" src="https://github.com/openai/evals/assets/27363885/7bf74e9a-c58e-414b-bb43-a52709d9437b">

When we send such a large request in a single batch, it could lead to several potential issues:

1. **Performance Impact**: Large HTTP requests could slow down the network and affect the performance of our application, especially if there are many such requests.
2. **Memory Usage**: The entire batch of events would need to be loaded into memory, which could lead to high memory usage if the batch size is very large.
3. **Request Failure**: The server receiving the request may have a maximum request body size limit. If the request exceeds this limit, it will be rejected.
4. **Timeouts**: If the request takes too long to send over the network, it may timeout, leading to failure of the request.

Therefore, having a reasonable limit on the `http_batch_size` could be a more balanced approach. It would help us strike a balance between reducing overhead and ensuring system stability. We can consider setting a default value that is sufficiently large to reduce overhead, but not so large that it risks causing the issues mentioned above.

I'm looking forward to hearing your thoughts on this.

How about I set it to something like `100`?

Reply:
100 as a default makes sense, thanks for the careful analysis!